server/index.py

```diff
@@ -0,0 +1,333 @@
+            if i == 2 or i == 3 or i == 4:
+                json_dict = json_dict | output_object
+
+        with open(json_path, 'w') as fp:
+            json.dump(json_dict, fp)
+
+            response = json_dict
+            return jsonify(response), 200
+    except Exception as e:
+        return process_error('failed to process file – check logs for details', e)
+
+def transcript_to_text(transcript):
+    text = " ".join([segment["text"] for segment in transcript])
+    return text
+
+@sieve.function(
+    name="video-dubber",
+    python_packages=[
+        "numpy>=1.19.0",
+    ],
+    system_packages=["ffmpeg"],
+)
+
+def video_dubbing(source_video: sieve.Video, language: str):
+    """
+    :param source_video: The video to dub
+    :param language: The language to dub the video in
+    :return: The dubbed video
+    """
+
+    print("extracting audio from video")
+    # Extract audio from video
+    source_audio = extract_audio(source_video)
+    print("done extracting")
+
+    # check if language is supported
+    if language not in ["english", "spanish", "french", "german", "italian", "portuguese", "polish", "turkish", "russian", "dutch", "czech", "arabic", "chinese"]:
+        raise Exception("Language not supported")
+
+    # use remote sieve models
+    transcriber = sieve.function.get("sieve/speech_transcriber:8eb4cdf")
+    translator = sieve.function.get("sieve/seamless_text2text:f8695f6")
+    tts = sieve.function.get("sieve/xtts-v1:e788931")
+    lipsyncer = sieve.function.get("sieve/video_retalking:de84f40")
+
+    print("transcribing audio")
+    # transcribe audio
+    transcript = list(transcriber.run(source_audio))
+    text = transcript_to_text(transcript)
+    text_language = "english"
+    print("transcription:", text)
+
+    print("translating text")
+    # Translate text
+    translated_text = translator.run(text, text_language, language)
+    print("translated text:", translated_text)
+
+    print("generating tts audio")
+    # Generate new audio from translated text
+    target_audio = tts.run(translated_text, source_audio, stability=0.5, similarity_boost=0.5)
+    print("done generating audio")
+
+    print("starting lipsync")
+    # Combine audio and video with Retalker
+    return lipsyncer.run(source_video, target_audio).path
+
+@app.route('/projects/<project_id>', methods=['GET'])
+async def get_status(project_id):
+    """Route to return current processing status of a project."""
+    status_file_name = f'{project_id}.txt'
+    status_file_path = os.path.join(get_output_dir_path(), status_file_name)
+    try:
+        with open(status_file_path, "r", encoding="utf-8") as f:
+            data = json.load(f)
+            return jsonify(data), 200
+    except Exception as e:
+        msg = "failed to open status file"
+        print(e, file=sys.stderr)
+        return jsonify({'error': msg}), 500
+
+
+@app.route('/projects/<project_id>/download', methods=['GET'])
+async def download_final_output(project_id):
```

Comment:
Same here - I don't see us writing anything to the provided output dir or returning a project ID to the user. I _do suggest_ using the configured output dir to store processing output vs a hard-coded vite path up above. But if you opt not to go this route and this actually is completely unused, I suggest removing all mentions of an output dir as well as this route (very possible I may be missing something somewhere though!)